sparql/SPARQLLexer.g4

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2023 by Bart Kiers
+ *
+ * The MIT license.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * (REVISION Last updated: 18 October 2023)
+ */
+lexer grammar SPARQLLexer;
+
+options { caseInsensitive = true; }
+
+SPACE : WS -> channel(HIDDEN);
+COMMENT : '#' ~[\r\n]* -> channel(HIDDEN);
+
+TRUE : 'TRUE';
+FALSE : 'FALSE';
+DISTINCT : 'DISTINCT';
+NOT : 'NOT';
+IN : 'IN';
+STR : 'STR';
+LANG : 'LANG';
+LANGMATCHES : 'LANGMATCHES';
+DATATYPE : 'DATATYPE';
+BOUND : 'BOUND';
+IRI : 'IRI';
+URI : 'URI';
+BNODE : 'BNODE';
+RAND : 'RAND';
+ABS : 'ABS';
+CEIL : 'CEIL';
+FLOOR : 'FLOOR';
+ROUND : 'ROUND';
+CONCAT : 'CONCAT';
+STRLEN : 'STRLEN';
+UCASE : 'UCASE';
+LCASE : 'LCASE';
+ENCODE_FOR_URI : 'ENCODE_FOR_URI';
+CONTAINS : 'CONTAINS';
+STRSTARTS : 'STRSTARTS';
+STRENDS : 'STRENDS';
+STRBEFORE : 'STRBEFORE';
+STRAFTER : 'STRAFTER';
+YEAR : 'YEAR';
+MONTH : 'MONTH';
+DAY : 'DAY';
+HOURS : 'HOURS';
+MINUTES : 'MINUTES';
+SECONDS : 'SECONDS';
+TIMEZONE : 'TIMEZONE';
+TZ : 'TZ';
+NOW : 'NOW';
+UUID : 'UUID';
+STRUUID : 'STRUUID';
+MD5 : 'MD5';
+SHA1 : 'SHA1';
+SHA256 : 'SHA256';
+SHA384 : 'SHA384';
+SHA512 : 'SHA512';
+COALESCE : 'COALESCE';
+IF : 'IF';
+STRLANG : 'STRLANG';
+STRDT : 'STRDT';
+SameTerm : 'SAMETERM';
+IsIRI : 'ISIRI';
+IsURI : 'ISURI';
+IsBLANK : 'ISBLANK';
+IsLITERAL : 'ISLITERAL';
+IsNUMERIC : 'ISNUMERIC';
+COUNT : 'COUNT';
+SUM : 'SUM';
+MIN : 'MIN';
+MAX : 'MAX';
+AVG : 'AVG';
+SAMPLE : 'SAMPLE';
+GROUP_CONCAT : 'GROUP_CONCAT';
+SEPARATOR : 'SEPARATOR';
+REGEX : 'REGEX';
+SUBSTR : 'SUBSTR';
+REPLACE : 'REPLACE';
+EXISTS : 'EXISTS';
+SELECT : 'SELECT';
+REDUCED : 'REDUCED';
+AS : 'AS';
+WHERE : 'WHERE';
+OPTIONAL : 'OPTIONAL';
+GRAPH : 'GRAPH';
+SERVICE : 'SERVICE';
+SILENT : 'SILENT';
+BIND : 'BIND';
+VALUES : 'VALUES';
+UNDEF : 'UNDEF';
+A_ : 'A';
```

Comment:
[Docs says](https://www.w3.org/TR/sparql11-query/#sparqlGrammar) `A` keyword is case-sensitive:

> Keywords are matched in a case-insensitive manner with the exception of the keyword 'a' which, in line with Turtle and N3, is used in place of the IRI rdf:type (in full, http://www.w3.org/1999/02/22-rdf-syntax-ns#type).